core/src/main/scala/org/ensime/indexer/DatabaseService.scala

```diff
@@ -28,83 +32,76 @@ class DatabaseService(dir: File) extends SLF4JLogging {
     (ds, Database.forDataSource(ds))
   }
 
-  def shutdown(): Unit = {
+  def shutdown()(implicit ec: ExecutionContext): Future[Unit] = for {
     // call directly - using slick withSession barfs as it runs a how many rows were updated
     // after shutdown is executed.
-    db.createSession().createStatement() execute "shutdown;"
-    datasource.close()
-  }
+    _ <- db.run(sqlu"shutdown")
+    _ <- db.shutdown
+    _ = datasource.close()
+  } yield ()
 
   if (!dir.exists) {
     log.info("creating the search database")
     dir.mkdirs()
-    db withSession { implicit s =>
-      (fileChecks.ddl ++ fqnSymbols.ddl).create
-    }
+    Await.result(
+      db.run(
+        (fileChecks.schema ++ fqnSymbols.schema).create
+      ),
+      DB_CREATE_TIMEOUT
```

Comment:
Given the way that ENSIME works, the end effect to the user is going to be the same whether the creation succeeds or fails. Therefore I'm tempted to remove the timeout. If ENSIME doesn't come up, the user will see that the latest log message is "creating the search database" and then no more processing. Perhaps we should have the response log message "... created the search database".